src/collection.ts

```diff
@@ -1687,26 +1698,23 @@ export class Collection implements OperationParent {
    * @param callback - An optional callback, a Promise will be returned if none is provided
    */
   group(
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
     keys: any,
-    condition: any,
-    initial: any,
-    reduce: any,
-    finalize: any,
-    command: any,
-    options: any,
-    callback: Callback
-  ) {
-    const args = Array.prototype.slice.call(arguments, 3);
-    callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
-    reduce = args.length ? args.shift() : null;
-    finalize = args.length ? args.shift() : null;
-    command = args.length ? args.shift() : null;
-    options = args.length ? args.shift() || {} : {};
+    condition: Document,
+    initial: Document,
+    // TODO: Use labeled tuples when api-extractor supports TS 4.0
+    ...args: [/*reduce?:*/ any, /*finalize?:*/ any, /*command?:*/ any, /*callback?:*/ Callback]
```

Comment:
I can leave the tuple here since most of the arguments are `any` its too permissive to error.